Externals/MayaDataModel/include/MayaDMBlendShape.h

```diff
@@ -29,31 +29,31 @@ class BlendShape : public GeometryFilter
 					inputComponentsTarget.write(file);
 				}
 			}* inputTargetItem;
+			size_t nbInputTargetItem;
 			float* targetWeights;
+			size_t nbTargetWeights;
 			void write(FILE* file) const
 			{
-				size_t size = sizeof(inputTargetItem)/sizeof(InputTargetItem);
-				for(size_t i=0; i<size; ++i)
+				for (size_t i = 0; i<nbInputTargetItem; ++i)
```

Comment:
nbInputTargetItem must be initialized by user, just like inputTargetItem pointer. FYI this part of the code is not used by OpenCOLLADA.